File: src/api/types/dashboard/traffic_map.rs
#[derive(Debug, Serialize)]
pub struct TrafficMapNodeResponse {
    pub(crate) id: String,
    pub(crate) name: String,
    pub(crate) region: String,
    pub(crate) role: String,
    pub(crate) lat: Option<f64>,
    pub(crate) lng: Option<f64>,
    pub(crate) country_code: Option<String>,
    pub(crate) country_name: Option<String>,
    pub(crate) geo_scope: String,
    pub(crate) traffic_weight: f64,
    pub(crate) request_count: u64,
    pub(crate) blocked_count: u64,
    pub(crate) bandwidth_mbps: f64,
    pub(crate) last_seen_at: i64,
}

#[derive(Debug, Serialize)]
pub struct TrafficMapFlowResponse {
    pub(crate) id: String,
    pub(crate) node_id: String,
    pub(crate) direction: String,
    pub(crate) decision: String,
    pub(crate) request_count: u64,
    pub(crate) bytes: u64,
    pub(crate) bandwidth_mbps: f64,
    pub(crate) average_latency_ms: u64,
    pub(crate) last_seen_at: i64,
}

#[derive(Debug, Serialize)]
pub struct TrafficMapResponse {
    pub(crate) scope: String,
    pub(crate) window_seconds: u32,
    pub(crate) generated_at: i64,
    pub(crate) runtime_pressure_level: String,
    pub(crate) degraded_reasons: Vec<String>,
    pub(crate) origin_node: TrafficMapNodeResponse,
    pub(crate) nodes: Vec<TrafficMapNodeResponse>,
    pub(crate) flows: Vec<TrafficMapFlowResponse>,
    pub(crate) active_node_count: u32,
    pub(crate) peak_bandwidth_mbps: f64,
    pub(crate) allowed_flow_count: u32,
    pub(crate) blocked_flow_count: u32,
    pub(crate) live_traffic_score: f64,
}

#[derive(Debug, Deserialize, Default)]
pub struct TrafficMapQueryParams {
    pub(crate) window_seconds: Option<u32>,
}
use serde::{Deserialize, Serialize};

File: dome-vue/src/features/dashboard/types/dashboard/trafficMap.ts
import type { SecurityEventItem } from '@/features/events/types/events'
import type { ApiQueryValue } from '@/shared/types/common'

export type EventMapScope = 'china' | 'global'

export interface TrafficMapQuery extends Record<string, ApiQueryValue> {
  window_seconds?: number
}
export interface EventMapNode {
  id: string
  name: string
  region: string
  role: 'cdn' | 'origin'
  lat?: number
  lng?: number
  countryCode?: string | null
  countryName?: string | null
  geoScope?: 'domestic' | 'global' | 'internal' | 'unknown' | string
  trafficWeight: number
  requestCount?: number
  blockedCount?: number
  bandwidthMbps?: number
  lastSeenAt?: number
}

export interface EventMapFlow {
  id: string
  nodeId: string
  direction: 'ingress' | 'egress'
  decision: 'allow' | 'block'
  intensity: number
  bandwidthMbps: number
  requestsPerSecond: number
  startedAt: number
  durationMs: number
  reason: string
  event?: SecurityEventItem
  requestCount?: number
  bytes?: number
  averageLatencyMs?: number
}

export interface EventMapSnapshot {
  scope: EventMapScope
  nodes: EventMapNode[]
  flows: EventMapFlow[]
  originNode: EventMapNode
  liveTrafficScore: number
  activeNodeCount: number
  peakBandwidthMbps: number
  allowedFlowCount: number
  blockedFlowCount: number
  hottestNode: EventMapNode | null
}

export interface TrafficMapNodeResponse {
  id: string
  name: string
  region: string
  role: 'cdn' | 'origin' | string
  lat?: number
  lng?: number
  country_code?: string | null
  country_name?: string | null
  geo_scope?: 'domestic' | 'global' | 'internal' | 'unknown' | string
  traffic_weight: number
  request_count: number
  blocked_count: number
  bandwidth_mbps: number
  last_seen_at: number
}

export interface TrafficMapFlowResponse {
  id: string
  node_id: string
  direction: 'ingress' | 'egress'
  decision: 'allow' | 'block'
  request_count: number
  bytes: number
  bandwidth_mbps: number
  average_latency_ms: number
  last_seen_at: number
}

export interface TrafficMapResponse {
  scope: EventMapScope
  window_seconds: number
  generated_at: number
  origin_node: TrafficMapNodeResponse
  nodes: TrafficMapNodeResponse[]
  flows: TrafficMapFlowResponse[]
  active_node_count: number
  peak_bandwidth_mbps: number
  allowed_flow_count: number
  blocked_flow_count: number
  live_traffic_score: number
}

export interface TrafficEventDeltaNode {
  id: string
  name: string
  region: string
  role: 'cdn' | 'origin' | string
  lat?: number
  lng?: number
  country_code?: string | null
  country_name?: string | null
  geo_scope?: 'domestic' | 'global' | 'internal' | 'unknown' | string
}

export interface TrafficEventDelta {
  timestamp_ms: number
  direction: 'ingress' | 'egress'
  decision: 'allow' | 'block'
  bytes: number
  latency_ms?: number | null
  source_ip: string
  node: TrafficEventDeltaNode
}
